controllers/drplacementcontrolvolsync.go

```diff
@@ -197,17 +201,9 @@ func (d *DRPCInstance) IsVolSyncReplicationRequired(homeCluster string) (bool, e
 		return false, fmt.Errorf("failed to find VRG on homeCluster %s", homeCluster)
 	}
 
-	if len(vrg.Status.ProtectedPVCs) == 0 {
```

Comment:
We need to rework this function. As we discussed, we can't wait for `ProtectedPVCs` to be filled in, but at the same time, we need to return `WaitForSourceCluster` error on an empty `ProtectedPVCs`, otherwise, we may complete the action (deploy/failover/relocate) without setting up volsync. The next reconcile may set it up, but we can't depend on that as we don't know when it would be called.

One possible solution is to return an `WaitForSouceCluster` error [here](https://github.com/RamenDR/ramen/blob/01d4e722c466d6a6949f5dd021a20200c648f6d9/controllers/drplacementcontrol.go#L1396) instead. Something like:

```
        vrg := d.vrgs[homeCluster]
	if len(vrg.Status.ProtectedPVCs) == 0 {
		return WaitForSourceCluster
	}
        repReq := d.IsVolSyncReplicationRequired(homeCluster)
	...
```
In other words, just move that block from `IsVolSyncReplicationRequired` and put it in `EnsureCleanup`. Also the `IsVolSyncReplicationRequired` return value becomes a boolean return value only.

Reply:
Thx Benamar, good catch - this should be addressed in the latest commit now: https://github.com/RamenDR/ramen/pull/529/commits/4229f1da9e644ace6039cfbf5f85bf9b14f8dac7